via/services/google_drive.py

```diff
@@ -0,0 +1,123 @@
+import json
+import re
+from io import BytesIO
+from json import JSONDecodeError
+from typing import ByteString, Iterator
+
+from google.oauth2.service_account import Credentials
+from googleapiclient.discovery import build
+from googleapiclient.errors import HttpError
+from googleapiclient.http import MediaIoBaseDownload
```

Comment:
I'll give this a shot in this case, as I think it might help. but I'm wary of making general advice from it.

It could a bit of a trap to look at a specific situation and extrapolate policy as the decision here hangs on a number of local decisions and judgement calls:

* It increases one type of readability at the expense of another. What's the exact cut off?
* This causes a problem with patching, but in this case we consider that an acceptable trade off. When is it too much?
* We only use them once. What if we used them twice, when's the cut off?
* The names are "generic". At which point do names become too generic? What makes a name good enough?
* This is 3rd party, but presumably `BytesIO` isn't?

The point isn't that the questions above don't have answers, they do, but the answers are situational. Which is to say, we may be dealing with a number of distinct situations which always require judgement rather than an opportunity for a general rule.

Having been refactoring about the code base a lot recently, I've seen this much more over used than under. If we had to pick a side of the coin for a simple rule (without exercising judgement) I'd recommend never doing it rather than always doing it. The bad effects I've seen are usually about the bloat you mention, which often pushes a single line onto two, which doesn't sound like a lot but can make things a lot longer and harder to read sometimes. 

The patching is also a bit weird. It seems nice to patch something out all in one go, but it can lead to some unwieldy assertions in the end. Pytests chains of calls and return values can already make somethings quite long:

```python
        MediaIoBaseDownload.return_value.next_chunk.assert_has_calls(...
```

If it's fully qualified that becomes:

```python
        googleapiclient.http.MediaIoBaseDownload.return_value.next_chunk.assert_has_calls(...
```

I think you can still sort of deep patch into the name space in the file to select different bits to get around that, but that would get us back to where we started with the tests.